falcon/request.py

```diff
@@ -2004,6 +2010,7 @@ def get_param_as_list(
         transform: Callable[[str], _T] | None = None,
         required: bool = False,
         store: StoreArg = None,
+        delimiter: str | None = None,
```

Comment:
Let's add the _delimiter_ parameter last. We haven't enforced keyword-only arguments in this method yet (maybe we should :thinking:), so this could otherwise theoretically be perceived as a breaking change if one passes everything as positional arguments.